### PR TITLE
refactor JDBC Project class to leverage new azkaban-db

### PR DESCRIPTION
**Pull request by kunkun-tang**
This change is aimed at re-constructing JDBC project loader class to
allow db retry feature to be leveraged. In order to make this happen,
there are a few changes made in this patch:
* remove some never-use abstract method in ProjectLoader interface.
* guicify JdbcProjectImpl and make it as the only implementation to
Projectloader.
* Seprate All Project related JDBC handler classes to a ourside static class.
* Adding getDataSource method in azkaban-db interface 

Evaluated the basic uses of ProjectLoader.

### CODE REVIEW

**Comment by kunkun-tang** [comment]
This PR is similar to #1040, which also tries leveraging newly constructed azkaban-db module.

**Comment by kunkun-tang** [comment]
rebased the latest code from master, and re-evaluated. Ready for review.

File: azkaban-common/src/main/java/azkaban/project/JdbcProjectHandlerSet.java [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+                resourceId);
+
+        handlers.add(handler);
+      } while (rs.next());
+
+      return handlers;
+    }
+  }
+
+  public static class IntHander implements ResultSetHandler<Integer> {
```
Comment (reallocf) [inline azkaban-common/src/main/java/azkaban/project/JdbcProjectHandlerSet.java]: IntHander or IntHandler?

File: azkaban-common/src/main/java/azkaban/project/JdbcProjectLoader.java [hunk 2]
```diff
@@ -785,7 +784,7 @@ public void removePermission(final Project project, final String name, final boo
 
   @Override
   public List<Triple<String, Boolean, Permission>> getProjectPermissions(
```
Comment (jamiesjc) [inline azkaban-common/src/main/java/azkaban/project/JdbcProjectLoader.java]: This method doesn't seem to be used. Could be removed later.

File: azkaban-common/src/main/java/azkaban/project/ProjectLoader.java [hunk 3]
```diff
@@ -200,7 +200,7 @@ void updateProjectProperty(Project project, Props props)
   Props fetchProjectProperty(int projectId, int projectVer, String propsName)
       throws ProjectManagerException;
 
-  List<Triple<String, Boolean, Permission>> getProjectPermissions(int projectId)
```
Comment (jamiesjc) [inline azkaban-common/src/main/java/azkaban/project/ProjectLoader.java]: Same comment as above.

File: azkaban-common/src/test/java/azkaban/project/MockProjectLoader.java [hunk 4]
```diff
@@ -231,7 +231,7 @@ public Props fetchProjectProperty(final int projectId, final int projectVer,
 
   @Override
   public List<Triple<String, Boolean, Permission>> getProjectPermissions(
```
Comment (jamiesjc) [inline azkaban-common/src/test/java/azkaban/project/MockProjectLoader.java]: Same comment as above.

File: azkaban-common/src/main/java/azkaban/project/JdbcProjectHandlerSet.java (hunk 1 above)
Comment (kunkun-tang) [inline azkaban-common/src/main/java/azkaban/project/JdbcProjectHandlerSet.java]: Good Catch! Changing!

File: azkaban-common/src/main/java/azkaban/project/JdbcProjectLoader.java (hunk 2 above)
Comment (kunkun-tang) [inline azkaban-common/src/main/java/azkaban/project/JdbcProjectLoader.java]: I searched and recalled that this method has to be used in my test. So I didn't remove it.

**Pull request merged**